src/call_builder.ts

```diff
@@ -242,6 +244,26 @@ export class CallBuilder<
     return this;
   }
 
+  /**
+   * A helper method to craft queries to "neighbor" endpoints.
+   *
+   *  For example, we have an `/effects` suffix endpoint on many different
+   *  "root" endpoints, such as `/transactions/:id` and `/accounts/:id`. So,
+   *  it's helpful to be able to conveniently create queries to the
+   *  `/accounts/:id/effects` endpoint:
+   *
+   *    this.forEndpoint("accounts", accountId)`.
+   *
+   * @param  {string} endpoint neighbor endpoint in question, like /operations
+   * @param  {string} param    filter parameter, like an operation ID
+   *
+   * @returns {CallBuilder} this CallBuilder instance
+   */
+  protected forEndpoint(endpoint: string, param: string): this {
+    this.filter.push([endpoint, param, this.neighborRoot]);
```

Comment:
WDYT of making this method throw an error when `this.neighborRoot` is not set? It'd make its usage safer IMO.

Alternatively, we could enforce the constructor to require a non-empty `neighborRoot` but I'm not sure that's desired for the SDK.

Reply:
Agreed on the`throw`, good call! Done in fd1c4af0.